app/src/main/java/com/portfolio/tasky/entry/fragments/LoginFragment.kt

```diff
@@ -1,56 +1,137 @@
 package com.portfolio.tasky.entry.fragments
 
 import android.os.Bundle
+import android.text.Editable
 import android.view.Gravity
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
 import androidx.fragment.app.Fragment
-import androidx.transition.Fade
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.viewModels
 import androidx.transition.Slide
-import com.portfolio.tasky.R
+import com.portfolio.tasky.*
 import com.portfolio.tasky.databinding.LayoutLoginBinding
 import com.portfolio.tasky.entry.EntryActivity
-import com.portfolio.tasky.FragmentInflater
-import com.portfolio.tasky.FragmentInflaterImpl
+import com.portfolio.tasky.viewModels.LoginViewModel
+import com.portfolio.tasky.views.TaskyAppCompatEditText
+import dagger.hilt.android.AndroidEntryPoint
 
-class LoginFragment : Fragment(), FragmentInflater by FragmentInflaterImpl() {
+@AndroidEntryPoint
+class LoginFragment : Fragment(), FragmentInflater by FragmentInflaterImpl(), TextChanged  {
     private lateinit var viewBinding: LayoutLoginBinding
 
+    private val viewModel: LoginViewModel by viewModels()
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        viewBinding = LayoutLoginBinding.inflate(inflater, container, false)
+        viewBinding = DataBindingUtil.inflate(inflater, R.layout.layout_login, container, false)
+        viewBinding.lifecycleOwner = this
         return viewBinding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        setObservers()
         viewBinding.btnLogin.setOnClickListener {
             openRegistrationFragment()
         }
     }
 
+    private fun setObservers() {
+        viewModel.emailChange.observe(viewLifecycleOwner){
+            val emailField = viewBinding.etEmail
+            it?.let { it1 -> emailField.setValid(it1)
+                emailField.setError(emailField.subLayout.etInput.text?.isNotEmpty() as Boolean && !it1)}
+        }
+        viewModel.passwordChange.observe(viewLifecycleOwner){
+            val passwordField = viewBinding.etPassword
+            it?.let { it1 -> passwordField.setValid(it1)
+                passwordField.setError(passwordField.subLayout.etInput.text?.isNotEmpty() as Boolean && !it1)}
+        }
+        viewModel.areFieldsValid.observe(viewLifecycleOwner){
+            it.let {  viewBinding.btnLogin.isEnabled = it == true }
+        }
+    }
+
+    private fun setFieldValidations(loginFragment: LoginFragment?) {
```

Comment:
Why is LoginFragment nullable?

I made another comment, maybe that fixes the nullability

Reply:
Some experimentation and then forgot to clean it up

Done